### PR TITLE
Issue #3311777 by hitesh-jain: Fix broken notifications to add account exists validation

### PR DESCRIPTION
**Pull request by hitesh-jain**
## Problem

Notifications are broken for the recipient users where the activity user owners are removed from the system manually causing notification loading issue on PHP 8.0 version.

`TypeError: Drupal\profile\ProfileStorage::loadByUser(): Argument #1 ($account) must be of type Drupal\Core\Session\AccountInterface, null given, called in /var/www/html/docroot/profiles/contrib/social/modules/custom/activity_creator/activity.page.inc on line 88 in Drupal\profile\ProfileStorage->loadByUser() (line 33 of /var/www/html/docroot/modules/contrib/profile/src/ProfileStorage.php)
`

## Solution
Add appropriate validation to check for User Account exists before trying to load that particular user account.

## Issue tracker
https://www.drupal.org/project/social/issues/3311777

### Before merge
- [ ] Code/peer review is completed
- [ ] All commit messages are [clear and clean](https://open-social.slite.com/app/docs/DnmermZDIx_0OQ). If applicable a rebase was performed
- [ ] All automated tests are green
- [ ] Functional/manual tests of the acceptance criteria are approved
- [ ] All acceptance criteria were met
- [ ] New features or changes to existing features are covered by tests, either unit (preferably) or behat
- [ ] Update path is tested. New hook_updates should respect update order, right naming convention and consider hook_post_update code
- [ ] Module can be safely uninstalled. Update/implement hook_uninstall and make sure that removed configuration or dependencies are removed/uninstalled
- [ ] This pull request has all required labels (team/type/priority)
- [ ] This pull request has a milestone
- [ ] This pull request has an assignee (if applicable)
- [ ] Any front end changes are tested on all major browsers
- [ ] New UI elements, or changes on UI elements are approved by the design team
- [ ] New features, or feature changes are approved by the product owner

### After merge
- [ ] Code is tested on all branches that it has been cherry-picked
- [ ] Update hook number might need adjustment, make sure they have the correct order
- [ ] The Drupal.org ticket(s) are updated according to this pull request status

## Screenshots
*[Required if new feature, and if applicable] If this Pull Request makes visual changes then please include some screenshots that show what has changed here. A before and after screenshot helps the reviewer determine what changes were made.*

## Release notes
*[Required if new feature, and if applicable] A short summary of the changes that were made that can be included in release notes.*

## Change Record
*[Required if applicable] If this Pull Request changes the way that developers should do things or introduces a new API for developers then a change record to document this is needed. Please provide a draft for a change record or a link to an unpublished change record below. Existing change records can be consulted as example. Please provide a draft for a change record or a link to an unpublished change record below. [Existing change records](https://www.drupal.org/list-changes/social) can be consulted as example.*

## Translations
*[Optional]Translatable strings are always extracted from the latest development branch. To ensure translations remain available for platforms running older versions of Open Social the original string should be added to `translations.php` when it's changed or removed.*
- [ ] Changed or removed source strings are added to the `translations.php` file.

### CODE REVIEW

**Review by navneet0693 — APPROVED** [review]

**Comment by hitesh-jain** [comment]
Rebased on top of latest 11.5.x

**Review by navneet0693 — APPROVED** [review]

**Comment by navneet0693** [comment]
🍒 - picked on main.

**Comment by navneet0693** [comment]
🍒 - picked to 11.4.x

File: modules/custom/activity_creator/activity.page.inc [hunk 1]
```diff
@@ -84,7 +84,7 @@ function template_preprocess_activity(array &$variables) {
   // 'compact_notification' view mode might not be configured, so remember to
   // always check the theme setting first.
   $storage = \Drupal::entityTypeManager()->getStorage('profile');
-  if ($storage !== NULL) {
+  if ($storage !== NULL && $account !== NULL) {
```
Comment (ronaldtebrake) [inline modules/custom/activity_creator/activity.page.inc:87]: Hi @navneet0693 
This causes an PHPStan issue in main, could you perhaps check that out?

Comment (navneet0693) [reply modules/custom/activity_creator/activity.page.inc:87]: @ronaldtebrake Sure, I will open PR for fixing it, I spotted it as well on https://github.com/goalgorilla/open_social/pull/3060 and https://github.com/goalgorilla/open_social/pull/2877

**Pull request merged**